ci: Propose a workflow that doesn't fail on modern Java and Gradle

NotMyFault opened the pull request:
This PR aims to:
- Modernize the build workflow to not fail on Java 17 (and higher) and Gradle 7.2
- Utilize the gradle nexus plugin in combination with GitHub actions to create a continuous integration to test commits and pull requests and publish snapshots to OSS Sonatype or releases to the gradle plugin portal based on the authentication thru [GitHub repository secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository). This is a one time setup.
- Appveyor and travis have been removed in favor of the previously named point, GitHub actions.

ppalaga commented:
Thanks for the contribution, @NotMyFault! I have removed the Appweyor and Travis hooks and added Sonatype and Gradle Plugin portal secrets. Could you plz. check whether https://github.com/ec4j/editorconfig-gradle-plugin/actions/runs/1277481314 does what it is supposed to do?

NotMyFault commented:
> Thanks for the contribution, @NotMyFault! I have removed the Appweyor and Travis hooks and added Sonatype and Gradle Plugin portal secrets. Could you plz. check whether https://github.com/ec4j/editorconfig-gradle-plugin/actions/runs/1277481314 does what it is supposed to do?

![](https://i.imgur.com/7Wm276r.png)
All tasks where completed fine, "publish release" didn't run, because the version set in the build.gradle.kts ends with `-SNAPSHOT`, so the artifact has been published to the snapshot repository.

> How can I actually trigger a release?

If you added your `GRADLE_PLUGIN_API_KEY` and `GRADLE_PLUGIN_API_SECRET` (matching the workflow name in build.yml), pushing a change that increases that removes `-SNAPSHOT`, e.g. 0.0.4, will trigger an update. Once the build completed, you can increase the version and return to `-SNAPSHOT`, eg. 0.0.5-SNAPSHOT and deployment returns to the snapshot repository.

Tho, I noticed I introduced a little typo in the workflow that listens for gh releases instead a pushes. I drafter another pull request to correct that.
